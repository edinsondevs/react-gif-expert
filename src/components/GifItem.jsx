export const GifItem = (props) => {
    const { title, image } = props.item;
    
  return (
    <div className="card" >
     <img src={ image } alt={ title } width='300px' height='300px'  
     />
     <h6>{ title }</h6>   
    </div>
  )
}
