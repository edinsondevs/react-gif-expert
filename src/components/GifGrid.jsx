import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
            { isLoading && ( <h3>Cargando...</h3> ) }
                <h3>{category}</h3>
                <div className='card-grid' data-testid="content-cards">
                    {images.map((item, id) => (
                        <GifItem
                            key={item.id}
                            title={item.title}
                            url={item.image}
                        />
                        ))}
                </div>
                
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}