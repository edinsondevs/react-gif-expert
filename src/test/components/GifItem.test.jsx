import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";


describe('pruebas en GifItem', () => {

    const title = "Saitama";
    const url = "https://one-punch.com/saitama.jpg";
    
    test('debe hacer match con snapshopt ', () => {

        const { container } =  render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });

    test('debe mostrar la imagen con el url y alt indicado', () => {
        
        render( <GifItem title={ title } url={ url } /> );
        // screen.debug()
        // console.log(screen.getByRole('img').src)
        expect(screen.getByRole("img").src).toBe( url );
    });

    test('verificar que el title exista', () => {
        
        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title )).toBeTruthy();
    });

});