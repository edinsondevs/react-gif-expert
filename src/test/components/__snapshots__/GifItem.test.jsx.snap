// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem debe hacer match con snapshopt  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      height="300px"
      src="https://one-punch.com/saitama.jpg"
      width="300px"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
